### PR TITLE
Preserve test environment for post-mortem debug [v3]

### PR DESCRIPTION
**Pull request by apahim**
v1: #1615 

v2: #1620 
- New option `--keep-tmp (on|off)` for Avocado run command with precedence over the `keep_tmp_files` config option.

v3:
- Drop non-working config option `keep_tmp_files`.
- Decide whether we will keep the tmp dir or not based on the `basedir`. If it's not `None`, then we create the tmp_dir in the `basedir` location and don't remove it when the job finishes.

### CODE REVIEW

**Comment by apahim** [comment]
[Preserve test environment for post-mortem debug](https://trello.com/c/Q6I65OZl/33-preserve-test-environment-for-post-mortem-debug)

File: avocado/core/data_dir.py [hunk 1]
```diff
@@ -212,7 +223,7 @@ def __del__(self):
 _tmp_tracker = _TmpDirTracker()
 
 
-def get_tmp_dir():
+def get_tmp_dir(basedir=None, keep_tmp=None):
```
Comment (ldoktor) [inline avocado/core/data_dir.py]: You forgot the `keep_tmp` here...

Comment (apahim) [inline avocado/core/data_dir.py]: o>

**Comment by ldoktor** [comment]
Actually as you already need to update the code, would you please improve the commit heading of the first patch? It should include the feature name...

**Comment by apahim** [comment]
@ldoktor, updated.

**Comment by ldoktor** [comment]
Thanks, works well...

**Pull request merged**